### PR TITLE
Senlin: Nodes Op

### PR DESCRIPTION
**Pull request by JackKuei**
Prior to a PR being reviewed, there needs to be a Github issue that the PR
addresses. Replace the brackets and text below with that issue number.

For #[823 [Add support for Senlin clustering]](https://github.com/gophercloud/gophercloud/issues/823)

Links to the line numbers/files in the OpenStack source code that support the
code in this PR:

[[nodes:op api]](https://github.com/openstack/senlin/blob/e5bc16b3fea2f10dead365f983c5a937f23d72b0/senlin/api/openstack/v1/nodes.py#L197)
[[nodes:op engine]](https://github.com/openstack/senlin/blob/e5bc16b3fea2f10dead365f983c5a937f23d72b0/senlin/engine/node.py#L394)
[[nova op names]](https://github.com/openstack/senlin/blob/e5bc16b3fea2f10dead365f983c5a937f23d72b0/senlin/profiles/os/nova/server.py#L241)
[[heat op names]](https://github.com/openstack/senlin/blob/e5bc16b3fea2f10dead365f983c5a937f23d72b0/senlin/profiles/os/heat/stack.py#L88)

### CODE REVIEW

**Comment by JackKuei** [comment]
@jtopjian This one is ready to be reviewed. Thx.

**Comment by JackKuei** [comment]
Yes, changing from OpOptsBuilder => OperationOptsBuilder would definitely make it clearer for the user. I was debating with that since all the other functions follows the pattern of OpOptsBuilder even though "OpOpts" definitely is very confusing. I'm OK with the wording changes.

**Comment by jtopjian** [comment]
Sounds good.

Yes, I think this is definitely a case that warrants breaking standard styles :)

**Comment by JackKuei** [comment]
As for the struct changes, it basically boils down to which one looks better.

Operation with params:
```
{ 
 Operation: nodes.RebootOperation, 
 Params:    nodes.OpParams{"type": "SOFT"}
},
```
OR
```
{
     nodes.RebootOperation: nodes.OpParams{
          "type": "SOFT",
     },
},
```

Operation without params:
```
{
   Operation: nodes.LockOperation
},
```
OR 
```
{
    nodes.LockOperation: nil,
}
```

**Comment by jtopjian** [comment]
Let's keep the existing structure.

The alternative format is useful because it more closely mirrors the JSON body, but it feels a little awkward in Go. I prefer to have an explicit field for the Operation.

**Comment by JackKuei** [comment]
Sounds good

File: openstack/clustering/v1/nodes/doc.go [hunk 1]
```diff
@@ -70,5 +70,19 @@ Example to Get a Node
...
+	nodeID := "node123"
+	operationOpts := nodes.OperationOpts{
+		Operation: nodes.RebootOperation,
+		Params:    nodes.OperationParams{"type": "SOFT"},
+	}
+    actionID, err := nodes.Ops(serviceClient, nodeID, operationOpts).Extract()
+	if err != nil {
+		panic(err)
+	}
+
```
Comment (jtopjian) [inline openstack/clustering/v1/nodes/doc.go]: @JackKuei I'm so sorry - I didn't see this before. Looks like there's a formatting issue here.

Other than that, this LGTM and I can merge once fixed.

Comment (JackKuei) [reply openstack/clustering/v1/nodes/doc.go]: Good catch! Making a fix

**Pull request merged**